packages/mds-agency/tests/integration-tests.spec.ts

```diff
@@ -1509,3 +1538,124 @@ describe('Tests Stops', async () => {
       })
   })
 })
+
+describe('Tests for taxi modality', async () => {
+  before(async () => {
+    await Promise.all([db.initialize(), cache.initialize()])
+  })
+
+  it('verifies post taxi success', done => {
+    request
+      .post(pathPrefix('/vehicles'))
+      .set('Authorization', AUTH)
+      .send(TEST_TAXI)
+      .expect(201)
+      .end((err, result) => {
+        done(err)
+      })
+  })
+
+  for (const taxiEvent of TAXI_VEHICLE_EVENTS) {
+    const validStates = TAXI_EVENT_STATES_MAP[taxiEvent]
+    for (const vehicle_state of validStates) {
+      it(`verifies ${taxiEvent} success`, done => {
+        const { device_id } = TEST_TAXI
+        let trip_id
+        if (taxiEvent.startsWith('trip_')) {
+          trip_id = '1f943d59-ccc9-4d91-b6e2-0c5e771cbc6b'
+        }
+        const body = {
+          event_types: [taxiEvent],
+          vehicle_state,
+          telemetry: TEST_TELEMETRY,
+          timestamp: now(),
+          trip_id
+        }
+        request
+          .post(pathPrefix(`/vehicles/${device_id}/event`))
+          .set('Authorization', AUTH)
+          .send(body)
+          .expect(201)
+          .end((err, result) => {
+            // test.string(result.body.status).is(EVENT_STATUS_MAP[taxiEvent])
+            done(err)
+          })
+      })
+    }
+  }
+
+  /* We want to test for all micromobility events which
+   * are not included in the valid Taxi events.
+   */
+  const MICRO_MOBILITY_EVENTS_NOT_IN_TAXI_EVENTS = MICRO_MOBILITY_VEHICLE_EVENTS.filter(
+    item => TAXI_VEHICLE_EVENTS.indexOf(item as TAXI_VEHICLE_EVENT) < 0
+  )
+
+  for (const microEvent of MICRO_MOBILITY_EVENTS_NOT_IN_TAXI_EVENTS) {
+    const validStates = MICRO_MOBILITY_EVENT_STATES_MAP[microEvent]
+    for (const vehicle_state of validStates) {
+      it('verifies cannot send micro-mobility type event for a taxi', done => {
+        const { device_id } = TEST_TAXI
+        request
+          .post(pathPrefix(`/vehicles/${device_id}/event`))
+          .set('Authorization', AUTH)
+          .send({
+            event_types: [microEvent],
+            vehicle_state,
+            telemetry: TEST_TELEMETRY,
+            timestamp: now()
+          })
+          .expect(400)
+          .end((err, result) => {
+            test.string(result.body.error).contains('bad_param')
+            test.string(result.body.error_description).contains('invalid event_type')
+            done(err)
+          })
+      })
+    }
+  }
+})
+
+describe('Tests TripMetadata', async () => {
+  const metadata: Omit<TripMetadata, 'provider_id'> = {
+    trip_id: uuid(),
+    reserve_time: now(),
+    dispatch_time: now(),
+    trip_start_time: now(),
+    trip_end_time: now(),
+    distance: 100,
+    accessibility_options_used: [],
+    fare: {
+      quoted_cost: 2000,
+      actual_cost: 2500,
```

Comment:
If it doesn't already exist, another test to verify that the payment adds up to only the `actual_cost` and not the `quoted_cost` would be good.

Reply:
Right now we will accept basically anything that's sent, and it'd be a technical compliance violation (post-hoc analysis) if their numbers don't align well. We _could_ enforce it at POST time in our object validation, but I worry we may lose some data that way.

Reply:
A'ight.